charts/trino/templates/deployment-worker.yaml

```
@@ -150,7 +166,24 @@ spec:
             failureThreshold: {{ .Values.worker.readinessProbe.failureThreshold | default 6 }}
             successThreshold: {{ .Values.worker.readinessProbe.successThreshold | default 1 }}
           lifecycle:
+            {{- if .Values.worker.lifecycle }}
+            {{- if .Values.worker.gracefulShutdown.enabled }}
+            {{- fail "The `worker.lifecycle` configuration conflicts with `worker.gracefulShutdown`. Either disable `worker.gracefulShutdown` (and apply the related configurations manually), or remove `worker.lifecycle`." }}
```

Small nitpick, avoid parentheses, ref: https://developers.google.com/style/parentheses
```suggestion
            {{- fail "The `worker.lifecycle` configuration conflicts with `worker.gracefulShutdown`. Either disable `worker.gracefulShutdown`, and apply the related configurations manually, or remove `worker.lifecycle`." }}
```

---

done!